tests/test_freqtradebot.py

```diff
@@ -1811,23 +1823,37 @@ def test_handle_stoploss_on_exchange_trailing_error(
             'stopPrice': '0.1'
         }
     }
+    trade.orders.append(
+        Order(
+            ft_order_side='stoploss',
+            ft_pair=trade.pair,
+            ft_is_open=True,
+            ft_amount=trade.amount,
+            ft_price=3,
+            order_id='abcd',
+            order_date=dt_now(),
+        )
+    )
     mocker.patch(f'{EXMS}.cancel_stoploss_order',
                  side_effect=InvalidOrderException())
     mocker.patch(f'{EXMS}.fetch_stoploss_order',
                  return_value=stoploss_order_hanging)
+    time_machine.shift(timedelta(minutes=50))
     freqtrade.handle_trailing_stoploss_on_exchange(trade, stoploss_order_hanging)
     assert log_has_re(r"Could not cancel stoploss order abcd for pair ETH/USDT.*", caplog)
 
     # Still try to create order
     assert stoploss.call_count == 1
+    # TODO: Is this actually correct ? This will create a new order every time,
+    assert len(trade.open_sl_orders) == 2
```

Comment:
@Axel-CH the thing i don't "really" like is this.

Essentially, the cancellation failed - yet we simply create a new order, resulting in 2 open stop orders.
It's not a new behavior either i think - but i'm not entirely sure what the correct approach is (why did cancelling fail? most likely, placing a new order will fail, too - as the amount was sold ...)

another, detailed look at the test (and the corresponding code, obviously) will definitely help.

Reply:
I think what is missing is the fact that "cancel_stoploss_on_exchange" was not updated . Imo it should cancel all stoploss on exchange orders. Could be replaced by a "cancel_all_stoploss_on_exchange" that will loop over all existing sl and maybe in the future by "cancel_last_stoploss_on_exchange" that will cancel only the last updated stoploss order in order to keep others sl orders.
Also, "cancel_stoploss_on_exchange" is still using and updating "trade.stoploss_order_id" wich should be replaced.

Reply:
If the cancellation failed, it could be because the order do not exist, in that case it is good to create a new one. It could also be because the order was already filled, in that case we should check if it is a partial fill, and if it is, create a new sl order for remaining quantity.

Reply:
well that's my point, yes
It's however not changed behavior  so i'm pretty uncertain if it's a good idea to look at this as part of this PR.
The only thing changing now is that we "know" that there's now 2 open SL orders, and won't "forget" the prior order ...

Reply:
Yes of course, I agree, as this PR is the first step to better stoploss handling it is not required to manage that edge case here

Reply:
yeah, so what's left is essentially "more tests" ...

(also, i'll not merge this before the next release, to get more time in testing this on a wider audience)

Reply:
Ok, by more tests you mean live or unit tests?

Reply:
live ... essentially testing the ACTUAL behavior, with the randomness of some calls just failing or whatever ... :laughing: 

Unit tests do work fine - but will only bring us that far :laughing:

Reply:
I will do live testing today on bybit

Reply:
Tested today stop-loss on exchange (bybit future):
- [x] Initial SL
- [x] Replace with custom stoploss
- [x] Remove on exchange by user / replaced by bot

I was not able to trigger an emergency exit, but i'm getting the same behavior on develop. No issue found on my side so far

Reply:
I also tested stop-loss managed in freqtrade and it worked fine, LGTM 👍